test/rest_api/test_task_manager.py

```diff
@@ -0,0 +1,123 @@
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert tasks, "task_status unregistered task that did not finish"
+
+def test_task_manager_status_done(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "sequence_number": 1, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+
+            resp = rest_api.send("POST", f"task_manager_test/finish_test_task/{task0}")
+            resp.raise_for_status()
+            resp = rest_api.send("GET", f"task_manager/task_status/{task0}")
+            resp.raise_for_status()
+
+            status = resp.json()
+            assert status["id"] == task0, "Incorrect task id"
+            assert status["state"] == "done", "Incorrect task state"
+            assert status["error"] == "", "Error message was set"
+            assert status["sequence_number"] == 1, "Incorrect task sequence number"
+            assert status["keyspace"] == "keyspace0", "Incorrect task keyspace"
+            assert status["table"] == "table0", "Incorrect task table"
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert not tasks, "task_status did not unregistered task that finished"
+
+def test_task_manager_status_failed(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "sequence_number": 1, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+
+            resp = rest_api.send("POST", f"task_manager_test/finish_test_task_with_error/{task0}")
+            resp = rest_api.send("GET", f"task_manager/task_status/{task0}")
+            resp.raise_for_status()
+
+            status = resp.json()
+            assert status["id"] == task0, "Incorrect task id"
+            assert status["state"] == "failed", "Incorrect task state"
+            assert status["error"] != "", "Error message was not set"
+            assert status["sequence_number"] == 1, "Incorrect task sequence number"
+            assert status["keyspace"] == "keyspace0", "Incorrect task keyspace"
+            assert status["table"] == "table0", "Incorrect task table"
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert not tasks, "task_status did not unregistered task that failed"
+
+def test_task_manager_not_abortable(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "sequence_number": 1, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+
+            resp = rest_api.send("POST", f"task_manager/abort_task/{task0}")
+            assert resp.status_code == 500, "Aborted unabortable task"
```

Comment:
The requests python module has symbolic definitions for the error codes iirc, better use those.
We should have examples in the current tests.